### PR TITLE
Fix #1420 Auto downloader should respect file pattern and propose correct filename

### LINKED ISSUE #1420: Auto downloader should respect file pattern and propose correct filename
Currently it just renames to `\bibtexkey`

### PR DESCRIPTION
**Pull request by stefan-kolb**
image

### CODE REVIEW

**Comment by simonharrer** [comment]
LGTM

File: src/main/java/net/sf/jabref/external/DownloadExternalFile.java [hunk 1]
```diff
@@ -267,7 +268,8 @@ private void downloadFinished() {
     }
     // FIXME: will break download if no bibtexkey is present!
     private String getSuggestedFileName(String suffix) {
-        String plannedName = bibtexKey == null ? "set-filename" : bibtexKey;
+        String plannedName = FileUtil.createFileNameFromPattern(databaseContext.getDatabase(), frame.getCurrentBasePanel().getSelectedEntries().get(0), Globals.journalAbbreviationLoader.getRepository());
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/external/DownloadExternalFile.java]: So the field `bibtexKey` is no longer required, right?

**Comment by tobiasdiez** [comment]
LGTM :+1:

File: src/main/java/net/sf/jabref/external/DownloadExternalFile.java (hunk 1 above)
Comment (stefan-kolb) [inline src/main/java/net/sf/jabref/external/DownloadExternalFile.java]: bibtexkey is used inside createFileNameFromPattern

**Pull request merged**